### PR TITLE
Add SPD endpoint tests

### PR DESCRIPTION
**Pull request by ruivieira**
Initial endpoint tests, starting with SPD.

### CODE REVIEW

File: explainability-service/src/main/java/org/kie/trustyai/service/endpoints/metrics/GroupStatisticalParityDifferenceEndpoint.java [hunk 1]
```diff
@@ -65,7 +61,13 @@ public Response spd(BaseMetricRequest request) throws DataframeCreateException {
 
         final Dataframe df = dataSource.getDataframe();
 
-        final double spd = calculator.calculateSPD(df, request);
+        final double spd;
+        try {
+            spd = calculator.calculateSPD(df, request);
+        } catch (MetricCalculationException e) {
+            LOG.error("Error calculating metric: " + e.getMessage());
```
Comment (danielezonca) [inline explainability-service/src/main/java/org/kie/trustyai/service/endpoints/metrics/GroupStatisticalParityDifferenceEndpoint.java]: Let's log the stacktrace for debugging
```suggestion
            LOG.error("Error calculating metric: " + e.getMessage(), e);
```

File: explainability-service/src/main/java/org/kie/trustyai/service/endpoints/metrics/MetricsCalculator.java [hunk 2]
```diff
@@ -10,24 +10,29 @@
...
+                    value -> value.equals(privilegedAttr));
+            final Value unprivilegedAttr = PayloadConverter.convertToValue(request.getUnprivilegedAttribute());
+            final Dataframe unprivileged = dataframe.filterByColumnValue(protectedIndex,
+                    value -> value.equals(unprivilegedAttr));
+            final Value favorableOutcomeAttr = PayloadConverter.convertToValue(request.getFavorableOutcome());
+            final Type favorableOutcomeAttrType = PayloadConverter.convertToType(request.getFavorableOutcome().getType());
+            return FairnessMetrics.groupStatisticalParityDifference(privileged, unprivileged,
+                    List.of(new Output(request.getOutcomeName(), favorableOutcomeAttrType, favorableOutcomeAttr, 1.0)));
+        } catch (Exception e) {
+            throw new MetricCalculationException(e.getMessage());
```
Comment (danielezonca) [inline explainability-service/src/main/java/org/kie/trustyai/service/endpoints/metrics/MetricsCalculator.java]:
```suggestion
            throw new MetricCalculationException(e.getMessage(), e);
```

**Review by tteofili — APPROVED** [review]
LGTM

File: explainability-service/src/test/java/org/kie/trustyai/service/endpoints/metrics/EndpointTestProfile.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+import io.quarkus.test.junit.QuarkusTestProfile;
+
+public class EndpointTestProfile implements QuarkusTestProfile {
+    @Override
+    public Map<String, String> getConfigOverrides() {
+        final Map<String, String> overrides = new HashMap<>();
+        overrides.put("service.storage-format", "PVC");
+        overrides.put("service.model-name", "example");
+        overrides.put("service.data-format", "CSV");
+        overrides.put("service.metrics-schedule", "5s");
```
Comment (danielezonca) [inline explainability-service/src/test/java/org/kie/trustyai/service/endpoints/metrics/EndpointTestProfile.java:15]: We can probably need to disable scheduler in a future PR in tests but for now it is fine

**Review by danielezonca — APPROVED** [review]

File: explainability-service/src/test/java/org/kie/trustyai/service/endpoints/metrics/EndpointTestProfile.java (hunk 3 above)
Comment (ruivieira) [reply explainability-service/src/test/java/org/kie/trustyai/service/endpoints/metrics/EndpointTestProfile.java:15]: Created #81

**Review by RobGeada — APPROVED** [review]
LGTM

**Pull request merged**